lib/paperclip/gif_transcoder.rb

```diff
@@ -7,7 +7,11 @@ class GifTranscoder < Paperclip::Processor
     def make
       num_frames = identify('-format %n :file', file: file.path).to_i
 
-      return file unless options[:style] == :original && num_frames > 1
+      unless options[:style] == :original && num_frames > 1
+        tmp_file = Paperclip::TempfileFactory.new.generate(attachment.instance.file_file_name)
+        tmp_file << file.read
+        return tmp_file
```

Comment:
The `IOError: closed` exception comes from Paperclip closing the file too early. After investigating this (thoughtbot/paperclip#875) the solution is to create an extra temp file to make sure it's open (I know, eugh)